client/client.go

```diff
@@ -32,6 +33,29 @@ import (
 	"github.com/canonical/pebble/internal/wsutil"
 )
 
+// SocketNotFoundError is the error type returned when the Pebble client fails
+// to find a unix socket at the specified path.
+type SocketNotFoundError struct {
+	// Error wrapped error.
+	Err error
+
+	// SocketPath of the non existent socket.
+	SocketPath string
+}
+
+// Error implements errors error interface.
+func (s SocketNotFoundError) Error() string {
+	if s.SocketPath == "" && s.Err != nil {
```

Comment:
Seems like this is dead code, as `s.Err` will always be non-nil and `s.SocketPath` will always be set.

Reply:
I was just trying to add the cases if something ever uses this error outside of our current use case. It was more just cover the case for when and if this happens.

Reply:
Fine with me. It's certainly not going to hurt. +1